src/Jackett/Services/IndexerManagerService.cs

```diff
@@ -25,70 +24,152 @@ public interface IIndexerManagerService
         IEnumerable<IIndexer> GetAllIndexers();
 
         void InitIndexers();
-        void InitCardigannIndexers(string path);
+        void InitCardigannIndexers(IEnumerable<string> path);
         void InitAggregateIndexer();
-        void SortIndexers();
     }
 
     public class IndexerManagerService : IIndexerManagerService
     {
-        private IContainer container;
-        private Logger logger;
-        private Dictionary<string, IIndexer> indexers = new Dictionary<string, IIndexer>();
         private ICacheService cacheService;
         private IIndexerConfigurationService configService;
+        private IProtectionService protectionService;
+        private IWebClient webClient;
+
+        private Logger logger;
+
+        private Dictionary<string, IIndexer> indexers = new Dictionary<string, IIndexer>();
         private AggregateIndexer aggregateIndexer;
 
-        public IndexerManagerService(IContainer c, IIndexerConfigurationService config, Logger l, ICacheService cache)
+        public IndexerManagerService(IIndexerConfigurationService config, IProtectionService protectionService, IWebClient webClient, Logger l, ICacheService cache)
         {
-            container = c;
             configService = config;
+            this.protectionService = protectionService;
+            this.webClient = webClient;
             logger = l;
             cacheService = cache;
         }
 
         public void InitIndexers()
         {
-            logger.Info("Using HTTP Client: " + container.Resolve<IWebClient>().GetType().Name);
-
-            foreach (var idx in container.Resolve<IEnumerable<IIndexer>>().OrderBy(_ => _.DisplayName))
+            logger.Info("Using HTTP Client: " + webClient.GetType().Name);
+
+            var ixs = new IIndexer[]{
```

Comment:
cardigann is preferred, but in certain cases (e.g. in case it's a gazelle based tracker) a code based implementation is easier or using cardiann is not possible at all.

For now I would make a common constructor mandatory.
I just updated AnimeTorrents in 0e4e9f42534180d1d90bdb9f5cc25852803ba31d (The workaround used before is no longer needed)